File: debug_redis.js
const redis = require('redis');
const { promisify } = require('util');

// The Redis URL from Heroku
const redisUrl = process.env.REDIS_URL || 'rediss://p5d8f287a9e023c96ba7e7d2ae233fcbbf55b6e022aa3ddb43e15a56e97e93ff2@ec2-3-82-214-32.compute-1.amazonaws.com:16470';

// Create Redis client with SSL options for rediss:// URLs
const redisOptions = {
  url: redisUrl,
  socket: {}
};

if (redisUrl.startsWith('rediss://')) {
  redisOptions.socket.tls = true;
  redisOptions.socket.rejectUnauthorized = false;
}

console.log('Connecting to Redis with options:', JSON.stringify(redisOptions, null, 2));

// Create the client
const client = redis.createClient(redisOptions);

// Set up error handler
client.on('error', (err) => {
  console.error('Redis Client Error:', err);
  process.exit(1);
});

// Connect and test
async function testRedis() {
  try {
    await client.connect();
    console.log('Connected to Redis successfully');
    
    const pingResult = await client.ping();
    console.log('Redis PING result:', pingResult);
    
    await client.quit();
    console.log('Redis connection closed');
  } catch (err) {
    console.error('Error connecting to Redis:', err);
    process.exit(1);
  }
}

testRedis(); 

File: debug_redis.rb
require 'redis'

# The Redis URL from Heroku (replace with your actual URL if needed)
redis_url = ENV['REDIS_URL'] || 'rediss://p5d8f287a9e023c96ba7e7d2ae233fcbbf55b6e022aa3ddb43e15a56e97e93ff2@ec2-3-82-214-32.compute-1.amazonaws.com:16470'

puts "Connecting to Redis at: #{redis_url}"

# Configure Redis with SSL if using rediss://
redis_options = { url: redis_url }

if redis_url.start_with?('rediss://')
  puts "Detected SSL Redis connection, enabling SSL options"
  require 'openssl'
  redis_options[:ssl] = true
  redis_options[:ssl_params] = { verify_mode: OpenSSL::SSL::VERIFY_NONE }
end

puts "Redis options: #{redis_options.inspect}"

# Create Redis client and test connection
begin
  redis = Redis.new(redis_options)
  result = redis.ping
  puts "Successfully connected to Redis! Response: #{result}"
rescue => e
  puts "Error connecting to Redis: #{e.message}"
  puts e.backtrace.join("\n")
end 